src/backend/plaf.rs

```diff
@@ -0,0 +1,285 @@
+                let plaf_fixed = ColumnFixed::new(column.annotation.clone());
+                self.add_id_index_mapping(column, c_column_id_to_p_column_index, fixed_index.unwrap());
+                plaf.columns.fixed.push(plaf_fixed);
+            }
+        }
+    }
+
+    fn convert_plaf_poly(&self, chiquito_poly: &cPolyExpr<F>, counter: &mut Counter) -> pExpr<PlonkVar> {
+        match chiquito_poly {
+            cPolyExpr::Const(c) => pExpr::Const(BigUint::from_bytes_le(&c.to_repr())), // PrimeField uses little endian for bytes representation
+            cPolyExpr::Sum(es) => {
+                let mut iter = es.iter();
+                let first = self.convert_plaf_poly(iter.next().unwrap(), counter);
+                iter.fold(first, |acc, e| acc + self.convert_plaf_poly(e, counter))
+            }
+            cPolyExpr::Mul(es) => {
+                let mut iter = es.iter();
+                let first = self.convert_plaf_poly(iter.next().unwrap(), counter);
+                iter.fold(first, |acc, e| acc * self.convert_plaf_poly(e, counter))
+            }
+            cPolyExpr::Neg(e) => -self.convert_plaf_poly(e, counter),
+            cPolyExpr::Pow(e, n) => {
+                if *n == 0 {
+                    pExpr::Const(BigUint::from(1u32))
+                } else {
+                    let e = self.convert_plaf_poly(e, counter);
+                    (1..*n).fold(e.clone(), |acc, _| acc * e.clone())
+                }
+            }
+            cPolyExpr::Halo2Expr(e) => pExpr::from(e),
+            cPolyExpr::Query(column, rotation, annotation) => {
+                let index = self.c_column_id_to_p_column_index.get(&column.uuid()).unwrap();
+                if self.debug {
+                    println!("GET c column id {} match p column index {}", column.uuid(), index);
+                    println!("MAP {:#?}", self.c_column_id_to_p_column_index);
+                }
+                pExpr::Var(PlonkVar::Query(self.convert_plaf_query(column, rotation, annotation, *index)))
+            }
+        }
+    }
+
+    fn add_id_index_mapping(&self, column: &cColumn, c_column_id_to_p_column_index: Option<&mut HashMap<u32, usize>>, counter: &mut Counter) {
+        match c_column_id_to_p_column_index {
+            Some(c_column_id_to_p_column_index) => {
+                c_column_id_to_p_column_index.insert(column.uuid(), counter.number());
+                if self.debug {
+                    println!("c column id {} match p column index {}", column.uuid(), counter.number());
+                }
+                counter.increment();
+            }
+            None => {}
+        }
+    }
+
+    fn convert_plaf_query(
+        &self,
+        column: &cColumn,
+        rotation: &i32,
+        _annotation: &String,
+        index: usize, // plaf index starts from 0 for each column type
+    ) -> ColumnQuery {
+        match column.ctype {
+            cAdvice | Halo2Advice => {
```

Comment:
@qwang98 for the same reason we cannot use the imported halo2 columns, we cannot also have queries to halo2 columns

Reply:
Fixed